### ISSUE TITLE
Uniform sending functions are causing uniform mix up?

### ISSUE THREAD

**Issue by slmjkdbtl** [grovesNL/glow#89]
```rust
unsafe fn uniform_2_f32(&self, location: Option<&Self::UniformLocation>, x: f32, y: f32) {
    let gl = &self.raw;
    gl.Uniform2f(*location.unwrap_or(&0) as i32, x, y);
}
```
some of my shader uniforms has location `Some(0)`, the direct behaviors are if I have a shader like this:
```rust
uniform vec2 u_resolution;
uniform vec2 u_mouse;

void main() {
    // drawing a circle using `u_resolution`, but doesn't use `u_mouse`
}
```
If I don't use the `u_mouse` in my shader, the location would be `None`, which becomes `0`, causing my actual uniform with location `Some(0)` to take the value I send to `u_mouse`, in this case `u_resolution`, resulting in flickering image. (for some reason it only mix up once in 2 frames, so every second frame the uniform would look right)

For a workaround I have to check if the location exists before sending:
```rust
for u in uniform_values {
    let loc = gl.get_uniform_location(program_id, u.name);
    if loc.is_some() {
        gl.uniform_2_f32(loc, u.x, u.y);
    }
}
```
to ensure all uniforms are in place.

**Comment by grovesNL** [comment]
Hi, thanks for reporting this issue!

What should be the expected behavior here? Should we only call `gl.Uniform2` when `location.is_some()`, or should we just panic when `location` is `None` (i.e. `location.expect("Invalid uniform location")`?

**Comment by slmjkdbtl** [comment]
In my experience it should not simply not send when uniform is not found
```rust
unsafe fn uniform_2_f32(&self, location: Option<&Self::UniformLocation>, x: f32, y: f32) {
    let gl = &self.raw;
    if let Some(loc) = location {
        gl.Uniform2f(loc as i32, x, y);  
    }
}
```
since there are cases when uniform is sent but they're not actually used in the shader. If you're ok with this I'll submit a PR

**Comment by grovesNL** [comment]
Sure, this sounds reasonable to me :+1: Could we update all the uniform functions to do this?

**Comment by slmjkdbtl** [comment]
👌 Yes all uniform functions should do the same. I just submitted a fix, but not so sure about the behavior on WebGL though as mentioned in the pr

**Comment by grovesNL** [comment]
Fixed by #90
